File: Python/07.py
 #
 # Crea un programa que invierta el orden de una cadena de texto
 # sin usar funciones propias del lenguaje que lo hagan de forma automática.
 # - Si le pasamos "Hola mundo" nos retornaría "odnum aloH"
 #
def invirtiendoCadenas(oracion):
    caracteres = []
    for i in range(len(oracion) - 1, -1, -1):
        caracteres.append(oracion[i])
    return ''.join(caracteres)

# PRUEBAS #
print(invirtiendoCadenas("Hola mundo"))
print(invirtiendoCadenas("Python"))
print(invirtiendoCadenas("12345"))
print(invirtiendoCadenas("A"))
print(invirtiendoCadenas(""))
print(invirtiendoCadenas("¿Cómo estás?"))
print(invirtiendoCadenas("   espacios   "))

File: C#/07.cs
/*
 * Crea un programa que invierta el orden de una cadena de texto
 * sin usar funciones propias del lenguaje que lo hagan de forma automática.
 * - Si le pasamos "Hola mundo" nos retornaría "odnum aloH"
 */
internal class InvirtiendoCadenas
{
    public string InvertirCadena(string cadena)
    {
        StringBuilder sb = new StringBuilder();
        for (int i = cadena.Length - 1; i >= 0; i--)
        {
            sb.Append(cadena[i]);
        }
        return sb.ToString();
    }
}
